### PR TITLE
[Search] 2023-10-01-preview API Review feedback fixes - update swagger to reflect api renaming

### PR DESCRIPTION
**Pull request by robertklee**
# Data Plane API - Pull Request
We realized that the swagger didn't have the correct naming for one of the properties, and was missing an extensible enum for the discriminator. Followup to #25862 

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Comment by robertklee** [comment]
I'd like to reference the same reasoning for this breaking change review as the previous PR. This API is not publicly yet and the swagger was incorrect when it was merged. 

image

**Comment by robertklee** [comment]
This is an example of our intended shape which is being deployed.

image

**Comment by ShivangiReja** [comment]
@robertklee Why [AzureOpenAIParameters.resourceUri](https://github.com/Azure/azure-rest-api-specs/blob/75383798f33fad4e968de9aee8fd12988aec0f89/specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json#L6388) is declared as string but not `uri`? Similarly for [AzureOpenAIEmbeddingSkill.resourceUri ](https://github.com/Azure/azure-rest-api-specs/blob/75383798f33fad4e968de9aee8fd12988aec0f89/specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json#L9459)and [CustomVectorizerParameters.uri](https://github.com/Azure/azure-rest-api-specs/blob/75383798f33fad4e968de9aee8fd12988aec0f89/specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json#L6428)? 

The generated code for these types treats them as `string` instead of `Uri`(see [here](https://github.com/Azure/azure-sdk-for-net/blob/76ed61f1666e8c2b2ad8e1a03ec871aec3946009/sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs#L420)).

**Comment by robertklee** [comment]
I will ask Carey

**Comment by robertklee** [comment]
@Careyjmac

**Comment by Careyjmac** [comment]
> @robertklee Why [AzureOpenAIParameters.resourceUri](https://github.com/Azure/azure-rest-api-specs/blob/75383798f33fad4e968de9aee8fd12988aec0f89/specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json#L6388) is declared as string but not `uri`? Similarly for [AzureOpenAIEmbeddingSkill.resourceUri ](https://github.com/Azure/azure-rest-api-specs/blob/75383798f33fad4e968de9aee8fd12988aec0f89/specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json#L9459)and [CustomVectorizerParameters.uri](https://github.com/Azure/azure-rest-api-specs/blob/75383798f33fad4e968de9aee8fd12988aec0f89/specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json#L6428)?
> 
> The generated code for these types treats them as `string` instead of `Uri`(see [here](https://github.com/Azure/azure-sdk-for-net/blob/76ed61f1666e8c2b2ad8e1a03ec871aec3946009/sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs#L420)).

Our API expects a string for these fields, though we do convert them to Uri in our domain models. Since the swagger should match our API, I think it makes sense to keep them as string here. If ya'll want to convert them to Uri in the SDK implementation that would make sense though. Note that a similar field also exists on the existing WebApiSkill (which is the skill version of the CustomVectorizerParameters).

**Review by heaths — CHANGES_REQUESTED** [review]
Could you fix the PoliCheck errors? Click details to see more. Might just need an exemption given this is old info.

That said, you'll need a breaking change signoff from @JeffreyRichter or @mikekistler. If the service version hasn't shipped yet, maybe it's a rubber stamp but they will say for sure. It's already in main, so the swagger has shipped, in essence.

**Comment by robertklee** [comment]
I think all our swaggers get flagged for the same policheck errors. :(

My PR didn't affect those strings. Is there any way to get those permanently exempted?

**Comment by ShivangiReja** [comment]
>Our API expects a string for these fields, though we do convert them to Uri in our domain models. Since the swagger should match our API, I think it makes sense to keep them as string here. If ya'll want to convert them to Uri in the SDK implementation that would make sense though. Note that a similar field also exists on the existing WebApiSkill (which is the skill version of the CustomVectorizerParameters).

@Careyjmac and @robertklee - I understand that these are string fields, but could we specify the `format: uri` to generate the `Uri` type for SDKs?

**Comment by robertklee** [comment]
Added @ShivangiReja

File: specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json
Comment (robertklee) [inline specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json:8252]: remove

Comment (robertklee) [reply specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json:8252]: checking with Bev about it right now

**Comment by heaths** [comment]
> My PR didn't affect those strings. Is there any way to get those permanently exempted?

@robertklee if you look at the checks summary and click the link, the error output should say something about exemptions. If not, check the root `CONTRIBUTING.md` file in this repo. Off hand, I don't know where the exemption doc lives but those are the places I'd check.

**Review by heaths — APPROVED** [review]

**Comment by robertklee** [comment]
> > My PR didn't affect those strings. Is there any way to get those permanently exempted?
> 
> @robertklee if you look at the checks summary and click the link, the error output should say something about exemptions. If not, check the root `CONTRIBUTING.md` file in this repo. Off hand, I don't know where the exemption doc lives but those are the places I'd check.

@BevLoh @sazeesha098 as a heads up

File: specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json
Comment (robertklee) [reply specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json:8252]: keeping as is since these are read operations, it is ok to set as required.

**Comment by ShivangiReja** [comment]
@robertklee, key fields like [VectorSearchVectorizer.name](https://github.com/Azure/azure-rest-api-specs/blob/0ec9d71b6bf3bfe939a6509dfb46821db9c86bc1/specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchservice.json#L6348-L6355) should be both required and readonly. Currently, I observe that the key fields are marked as required. Can we also include `"readOnly": true` for them?"

**Comment by robertklee** [comment]
@weidongxu-microsoft could you please help merge this PR? I believe I've received the necessary signoffs.

**Comment by robertklee** [comment]
@weidongxu-microsoft or @heaths if the PR looks good, could you or someone with write access please help us merge it? Thank you!

**Comment by mattmsft** [comment]
@heaths  please let us know if you can merge the PR

**Pull request merged**
